ext/json/json.c

```diff
@@ -178,6 +179,30 @@ PHP_JSON_API zend_result php_json_decode_ex(zval *return_value, const char *str,
 }
 /* }}} */
 
+/* {{{ */
+PHP_JSON_API zend_result php_is_json_ex(zval *return_value, const char *str, size_t str_len, zend_long options, zend_long depth)
+{
+	php_json_parser parser;
+
+	options |= PHP_JSON_VALIDATE_ONLY;
+	
+	php_json_parser_init(&parser, return_value, str, str_len, (int)options, (int)depth);
```

Comment:
You should implement this differently and a different `php_json_parser_methods` that you can pass `php_json_parser_init_ex`. That won't impact the current parsing speed and will be cleaner.

Reply:
Ok, I will give it another try on this, I ping you back when ready.

Reply:
@bukka I need support debugging this.

I was moving towards your suggestion, basically by passing NULL to every callback function used by the parser, and because of that I also added some checks in the parser itself.

I have a problem only on FreeBSD, for this test:
ext/json/tests/is_json_002.phpt

```php
try {
  is_json("-", 512, JSON_THROW_ON_ERROR);
} catch (Exception $exception) {
  echo $exception->getMessage() . PHP_EOL;
  var_dump(
    json_last_error(),
    json_last_error_msg()
  );
}
```

When I set the JSON_THROW_ON_ERROR flag.

![image](https://user-images.githubusercontent.com/25756860/185637744-91726e46-6914-4b12-97eb-8be5133d2a10.png)

![image](https://user-images.githubusercontent.com/25756860/185637844-e928899e-14e1-43ac-8810-c2921933154a.png)

![image](https://user-images.githubusercontent.com/25756860/185637936-beb49cde-9966-46ba-8ab6-430576da0837.png)

![image](https://user-images.githubusercontent.com/25756860/185638106-77030c29-7c4c-40c0-b4e3-5edd43df3cf8.png)


The problems in the Azure pipeline are related to something else reported by me also in (https://github.com/php/php-src/pull/9359) so we can ignore it.

What is going on? Can you help me please?

Reply:
I corrected my previous comment

Reply:
I still have a bug I Need to work out

Reply:
Almost there

Reply:
@bukka ready for review again (I will still keep the PR as a Draft). Thanks in advance.

(I can squash everything to 1 commit later, for me is important to keep them during development :smile: )

Reply:
I would probably define dummy function rather than setting it to NULL to not impact json_decode but guess it's not a big issue really. I think it's good enough for the RFC.

It should be probably also renamed to `json_` something (e.g. `json_validate`) as it is PHP extension convention - all functions should be prefixed with the extension name in this case: See point 7 in https://github.com/php/php-src/blob/master/CODING_STANDARDS.md#user-functionsmethods-naming-conventions

Reply:
Thanks a lot for the feedback. 

I Will rename the function. 

Also I Will try to do the suggested changes for the dummy function instead of Null 💪